velox/common/caching/tests/CachedFactoryTest.cpp

```diff
@@ -225,3 +225,33 @@ TEST(CachedFactoryTest, retrievedCached) {
   }
   EXPECT_EQ(*generated, 5);
 }
+
+TEST(CachedFactoryTest, disableCache) {
+  auto generator = std::make_unique<DoublerGenerator>();
+  auto* generated = &generator->generated_;
+  CachedFactory<int, int, DoublerGenerator> factory(
+      nullptr, std::move(generator));
+
+  auto val1 = factory.generate(1);
+  EXPECT_EQ(val1, cacheMiss(2));
+  EXPECT_EQ(*generated, 1);
+
+  auto val2 = factory.generate(1);
+  EXPECT_EQ(val2, cacheMiss(2));
+  EXPECT_EQ(*generated, 2);
+
+  EXPECT_EQ(factory.currentSize(), 0);
```

Comment:
Sorry, I didn't mean add empty lines between each line of code, only between logically separate test cases.